### PR TITLE
log vendored commit ids to a file

### PR DESCRIPTION
**Pull request by kr**

### CODE REVIEW

**Comment by kr** [comment]
PTAL

File: vendor.go [hunk 1]
```diff
@@ -101,6 +112,28 @@ func reportExtVendoredDep() {
...
+	if err != nil {
+		return err
+	}
+	w := bufio.NewWriter(f)
+	for ip, p := range manifest {
+		commit, err := commitHash(p.Dir)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%s: commit hash: %v", ip, err)
+		}
+		fmt.Fprintf(w, "%s\t%s\n", commit, ip)
```
Comment (bmizerany) [inline vendor.go]: Why the commit first? Alignment? :)

Comment (kr) [inline vendor.go]: yeah, alignment :)

File: vendor.go [hunk 2]
```diff
@@ -101,6 +112,28 @@ func reportExtVendoredDep() {
...
+func reportManifest(name string) error {
+	f, err := os.OpenFile(name, os.O_RDWR|os.O_CREATE|os.O_APPEND, 0666)
+	if err != nil {
+		return err
+	}
+	w := bufio.NewWriter(f)
+	for ip, p := range manifest {
+		commit, err := commitHash(p.Dir)
+		if err != nil {
+			fmt.Fprintf(os.Stderr, "%s: commit hash: %v", ip, err)
```
Comment (kr) [inline vendor.go]: oops this should be log.Printf

**Comment by kr** [comment]
PTAL

**Comment by bmizerany** [comment]
Regarding our out-of-band conversation: I found a few ways we can get the vendor log into a commit message. They are:

**Tell users to use `git commit -F vendor-commit-message`** - This seems to be the simplest way but users have to remember to include the vendor log in their messages. They could also `:r vendor-log` today from vim or their editors equivalent when writing the commit message.

**Install a .git/hooks/prepare-commit-msg hook** - This would be ideal to use in combination with `vendor-log` because it can append the log automatically if ./vendor has been updated.

Thoughts?

**Pull request merged**